pyiceberg/io/fsspec.py

```diff
@@ -176,14 +186,50 @@ def _gs(properties: Properties) -> AbstractFileSystem:
 def _adlfs(properties: Properties) -> AbstractFileSystem:
     from adlfs import AzureBlobFileSystem
 
+    for property_name in properties:
+        if property_name.startswith(DEPRECATED_ADLFS_PREFIX):
+            deprecated(
+                deprecated_in="0.7.0",
+                removed_in="0.8.0",
+                help_message=f"The property {property_name} is deprecated. Please use properties that start with adls.",
+            )(lambda: None)()
+
     return AzureBlobFileSystem(
-        connection_string=properties.get(ADLFS_CONNECTION_STRING),
-        account_name=properties.get(ADLFS_ACCOUNT_NAME),
-        account_key=properties.get(ADLFS_ACCOUNT_KEY),
-        sas_token=properties.get(ADLFS_SAS_TOKEN),
-        tenant_id=properties.get(ADLFS_TENANT_ID),
-        client_id=properties.get(ADLFS_CLIENT_ID),
-        client_secret=properties.get(ADLFS_ClIENT_SECRET),
+        connection_string=PropertyUtil.get_first_property_value(
```

Comment:
[nit] For `TableProperties` and `PropertyUtil`, we usually do inline import:
https://github.com/apache/iceberg-python/blob/4fc75891b84b8511a67df47894b2137feb2c4733/pyiceberg/io/fsspec.py#L128-L132